clientv3/integration/black_hole_test.go

```diff
@@ -26,8 +26,52 @@ import (
 	"go.etcd.io/etcd/integration"
 	"go.etcd.io/etcd/pkg/testutil"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
+func TestServerGRPCKeepAliveTimeout(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{
+		Size:                  2,
+		GRPCKeepAliveInterval: time.Second,
+		GRPCKeepAliveTimeout:  time.Second,
+	})
+	defer clus.Terminate(t)
+
+	eps := []string{clus.Members[0].GRPCAddr()}
+
+	ccfg := clientv3.Config{
+		Endpoints: []string{eps[0]},
+	}
+
+	cli, err := clientv3.New(ccfg)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cli.Close()
+
+	if _, err = clus.Client(1).Put(context.TODO(), "foo", "bar"); err != nil {
+		t.Fatal(err)
+	}
+	clus.Members[1].Blackhole()
+	time.Sleep(10 * time.Second)
+	// remove blackhole but connection should be unavailable now
```

Comment:
`// remove blackhole but keepalive ping should trigger server to close server-to-client connection`?

Reply:
Sure, that's a better detail.